docs/reference-guide/modules/events/pages/event-handlers.adoc

```diff
@@ -14,89 +14,395 @@ If the event handler does not need access to the payload of the message, you can
+When you evolve your event structures, handlers can often adapt by simply changing their parameter type, rather than requiring upcasters to transform stored events.
+
+[[processing-context]]
+== `ProcessingContext` in event handlers
+
+The `ProcessingContext` is a critical component in Axon 5.
+On the handling side, a `ProcessingContext` is **always** created by the framework when processing an event.
+
+=== Injecting `ProcessingContext`
+
+You can inject the `ProcessingContext` as a parameter in your event handler:
+
+[source,java]
+----
+@EventHandler
+public void handle(OrderPlacedEvent event, ProcessingContext context) {
+    // Access the message
+    Message message = Message.fromContext(context);
+
+    // Access metadata
+    String userId = message.metadata().get("userId");
+
+    // Register cleanup actions
+    context.onCleanup(ctx -> cleanupResources());
+
+    // Access correlation data
+    String correlationId = message.metadata().get("correlationId");
 }
+----
+
+=== Passing `ProcessingContext` to components
+
+When your event handler needs to dispatch commands, publish events, or interact with other Axon components, you should pass the `ProcessingContext` to those components.
+This ensures correlation data flows through your system.
 
-public class SubListener extends TopListener {
+**Note:** Some components like `CommandDispatcher` and `EventAppender` are automatically scoped to the current `ProcessingContext` when injected as handler parameters.
+However, you still need the `ProcessingContext` parameter to:
+- Pass context to your own services and repositories
+- Register lifecycle callbacks
+- Access resources or metadata
+
+[source,java]
+----
+@Component
+public class OrderEventHandler {
+
+    private final NotificationService notificationService;
 
     @EventHandler
-    public void handle(EventB event) {
+    public void handle(OrderPlacedEvent event,
+                       ProcessingContext context,
+                       CommandDispatcher commandDispatcher) {
+        // CommandDispatcher is automatically context-aware (no need to pass context)
+        commandDispatcher.send(new ReserveInventoryCommand(event.getOrderId()));
+
+        // Pass context to your own services that need correlation data
+        notificationService.sendOrderConfirmation(event, context);
+
+        // Use context for lifecycle callbacks
+        context.onCommit(ctx -> {
+            // This runs only after successful commit
+            auditLog.recordOrderPlaced(event.getOrderId());
+        });
+
+        // Access correlation data for logging
+        Message message = Message.fromContext(context);
+        String traceId = message.metadata().get("traceId");
+        logger.info("Processing order {} with trace {}", event.getOrderId(), traceId);
     }
+}
+----
+
+For more details on `ProcessingContext`, see xref:messaging-concepts:processing-context.adoc[ProcessingContext].
 
+[[parameter-injection]]
+== Parameter injection
```

Comment:
I think there is no need for the `== Parameter injection` section here, per my "supported parameter types" comment earlier.